src/item.cpp

```diff
@@ -3303,11 +3297,21 @@ int item::spoilage_sort_order()
 
 void item::calc_rot( const tripoint &location )
 {
+    // avoid needlessly calculating already rotten things
+    if( rotten() ) {
+        return;
+    }
```

Comment:
Can you double check id this does not collide with has_rotten_away() that iirc needs relative rot to go beyond rotten() state?

Reply:
Good call, I think that is a problem, forgot about that.  I still think there should be a barrier here, but will fix.